lib/nxt_support/util/crystalizer.rb

```diff
@@ -0,0 +1,34 @@
+module NxtSupport
+  class Crystallizer
+    Error = Class.new(StandardError)
+
+    include NxtInit
+
+    attr_init :collection, on_uniqueness_violation: -> { uniqueness_violation_handler }, attribute: :itself
```

Comment:
`with` instead of `attribute` might be also a good name candidate.

```
NxtSupport::Crystallizer.new(collection: collection, with: :name)
```

Reply:
I think I prefer something like attribute or method.

Reply:
I also like `with` a bit more but don't care too much.

Reply:
Alternatively we could pass a block or a proc instead of attribute for more flexibility.

If I understood it correctly, it's basically `Array#uniq` and `ActiveRecord#first!` combined. And it could as well be a method.

```ruby
users.ensure_uniq!(&:name)
users.ensure_uniq! { |user| user.address.town }
```